packages/esm-reports-app/src/components/overlay.component.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Button, Header } from '@carbon/react';
+import { ArrowLeft, Close } from '@carbon/react/icons';
+import { useLayoutType } from '@openmrs/esm-framework';
+import { closeOverlay, useOverlay } from '../hooks/useOverlay';
+import styles from './overlay.scss';
+
+const Overlay: React.FC = () => {
+  const { header, component, isOverlayOpen } = useOverlay();
+  const layout = useLayoutType();
+  const overlayClass = layout !== 'tablet' ? styles.desktopOverlay : styles.tabletOverlay;
+  return (
+    <>
+      {isOverlayOpen && (
+        <div className={overlayClass}>
+          {layout === 'tablet' && (
+            <Header onClick={() => closeOverlay()} aria-label="Tablet overlay" className={styles.tabletOverlayHeader}>
+              <Button hasIconOnly>
```

Comment:
Prefer [IconButton](https://react.carbondesignsystem.com/?path=/docs/components-iconbutton--overview) for icon-only buttons.

Reply:
fixed